microprofile/WFLY_13129_Microprofile_CLI_Script_Example.adoc

```diff
@@ -0,0 +1,95 @@
+= Example CLI script to evolve standalone configurations with microprofile
+:author:            JF Denise
+:email:             jdenise@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+We are proposing here to define a new script in docs/examples/enable-microprofile.cli
+The script is an “example” script to be applied on default standalone configurations. 
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-13129
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+
+[ ] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* The script is packaged in the galleon-pack maven module. It is a default package installed when WildFly is installed.
+* Installation path is _<JBOSS_HOME>/docs/examples/enable-microprofile.cli_.
+* To launch the script: _bin/jboss-cli.sh --file=docs/examples/enable-microprofile.cli_
+* By default it applies to standalone.xml, use _-Dconfig=<standalone-full.xml|standalone-ha.xml|standalone-full-ha.xml>_
+to apply to other standalone configurations.
+* The script informs the user with high level information on the modifications it applies to the configuration it is updating.
+* The script is expected to adapt to the differences that exist between various standalone configurations.
+* The script detects that a microprofile subsystem is already present in the config and skips it. 
+* The script secures the configuration using elytron security domain ApplicationDomain, http-authentication-factory and sasl-authentication-factory for items present in default configurations:
+** core-service/management
+*** access/identity security domain
+*** managementItf/http-upgrade sasl factory
+*** managementItf/ http auth factory
+** remoting/http-connector sasl factory
+** undertow 
+*** other application-domain
+*** http-invoker UNTOUCHED, references legacy security realm, no http-auth factory to set.
+** ejb3 other application-domain
+** jberet security domain
+** messaging-activemq server security-domain(full only)
+* The script remove the security subsystem
```

Comment:
On removal we should double check if any JASPIC or JACC updates are needed in the Elytron subsystem, maybe not but the present configuration I think may be lenient to allow it to co-exist with PicketBox - that leniency would no longer be required.

Reply:
@darranl , is there something required on the script front to allow for this removal?

Reply:
I think it is just a point to double check once we have the script, if there is anything it will only be small.

Reply:
@darranl Can you share a hint how to easily check this please?